[OUDS] Add "Font" tokens, utilities and documentation (#2774)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/content/docs/0.0/helpers/icon.md

---
layout: docs
title: Icon

> description: Quickly see the icon sizes you should use with a specific typography.

Reviewer: ```suggestion
description: Use one of the right icon sizes with a specific typography.
```

group: helpers
aliases:
  - "/docs/helpers/icon/"
toc: true
---

> ## Core concept

Reviewer: I find this title quite different of what we usually have. 
Maybe "About icon sizes"?

The icon classes are quite easy to use. You can use these classes upon `<img>`, `<svg>` or even icon font wherever the icon is. Please make sure to either the rules described in this page or follow the design.

Every icon class presented in here has the common following code:

{{< scss-docs name="ouds-font-icon-common" file="scss/helpers/_icon.scss" >}}

> ## Displays

Author: My logic was a descending order in the page but I can mix it if you prefer.

| `.bm-short-icon` | `1rem` (16px) | `1rem` (16px) | `1.25rem` (20px) |
{{< /bs-table >}}

</details>

{{< example >}}
<h1> <!-- h1 uses heading-xlarge font reference so all related classes as `.h1`, `.fs-hxl`, etc... -->
  <svg class="hxl-tall-icon me-short text-info" width="1rem" height="1rem" fill="currentColor" aria-hidden="true">
    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#vector"/>
  </svg>
  H1 with tall icon
</h1>
<h1> <!-- h1 uses heading-xlarge font reference so all related classes as `.h1`, `.fs-hxl`, etc... -->
  <svg class="hxl-medium-icon me-short text-info" width="1rem" height="1rem" fill="currentColor" aria-hidden="true">
    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#vector"/>
  </svg>
  H1 with medium icon
</h1>
<h1> <!-- h1 uses heading-xlarge font reference so all related classes as `.h1`, `.fs-hxl`, etc... -->
  <svg class="hxl-short-icon me-short text-info" width="1rem" height="1rem" fill="currentColor" aria-hidden="true">
    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#vector"/>
  </svg>
  H1 with short icon

> </h1>

Author: Question asked to change the naming of tokens.

> </h1>

Author: The naming will changed to `-lg` and `-md` and `-sm`. So classes will be named `.hxl-lg-icon` for example. Would you be fine with it or do we keep on our side `-large`, `-medium` and `-smal` ?

> </h1>

Reviewer 2: IMO, it is confusing with breakpoint names. Usually, `sm`, `md` and `lg` are used to express something changing according to viewport size. Even if I know there is `btn-sm` for small button and not for small screen. Maybe the name is clear enough to understand it means small icons and not for small screen.

    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#vector"/>
  </svg>
  <p class="pt-smash">

>     Default long paragraph with tall icon. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent fermentum diam consequat, fermentum sem nec, malesuada eros. Praesent eu lectus ligula. Nullam et est in augue pharetra ultricies quis non orci.

Reviewer 3: If you want to avoid adding all the Latin words in CSpell, maybe you can find a content in English 😄
